tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerVertex.java

```diff
@@ -40,7 +40,7 @@
 /**
  * @author Marko A. Rodriguez (http://markorodriguez.com)
  */
-public final class TinkerVertex extends TinkerElement implements Vertex {
```

Comment:
Could you please talk about the cases where folks would extend `TinkerVertex` (or `TinkerGraph` more generally)? We have one very specific one with https://github.com/ShiftLeftSecurity/tinkergraph-gremlin but i'm not sure that's relevant anymore with its successor in https://github.com/ShiftLeftSecurity/overflowdb

Reply:
It could be extended by vendors to create a custom testing suite OR it could be used by vendors to build a TinkerGraph++ offering. IMO, we should probably relax the final requirement on classes unless it is necessary or has some specific purpose.

Reply:
I just sorta realized this issue is still hanging around here without any added reply. I did start a discussion on this on dev and didn't really get objections but I'd prefer to do this particular TinkerGraph change around a JIRA (with a bit more thought as to the repercussions. If you just back out this one change then the rest of this is good to merge from my perspective. If you agree I will create a JIRA for the TinkerGraph issue. wdyt?

Reply:
Sure. I will move the TinkerGraph out of the scope of this PR as well so that things become simpler.

Reply:
I have removed changes in TinkerGraph from this PR and this is ready for review.